tests/client-sdk/agents/test_agents.py

```diff
@@ -267,3 +233,97 @@ def test_custom_tool(llama_stack_client, agent_config):
     logs_str = "".join(logs)
     assert "-100" in logs_str
     assert "CustomTool" in logs_str
+
+
+def test_rag_agent(llama_stack_client, agent_config):
+    urls = [
+        "memory_optimizations.rst",
+        "chat.rst",
+        "llama3.rst",
+        "datasets.rst",
+        "qat_finetune.rst",
+        "lora_finetune.rst",
+    ]
+    documents = [
+        Document(
+            document_id=f"num-{i}",
+            content=f"https://raw.githubusercontent.com/pytorch/torchtune/main/docs/source/tutorials/{url}",
+            mime_type="text/plain",
+            metadata={},
+        )
+        for i, url in enumerate(urls)
+    ]
+    llama_stack_client.memory_banks.register(
+        memory_bank_id="test_bank",
+        params={
+            "memory_bank_type": "vector",
+            "embedding_model": "all-MiniLM-L6-v2",
+            "chunk_size_in_tokens": 512,
+            "overlap_size_in_tokens": 64,
+        },
+        provider_id="faiss",
+    )
+
+    # insert some documents
+    llama_stack_client.memory.insert(
+        bank_id="test_bank",
+        documents=documents,
+    )
+
+    # create the required memory tool
+    llama_stack_client.toolgroups.register(
+        tool_group_id="memory_group",
+        tool_group=UserDefinedToolGroupDef(
+            type="user_defined",
+            tools=[
+                CustomToolDefParam(
+                    type="custom",
```

Comment:
Should we add a new type other than "custom" to distinguish between custom tools that gets executed on client v.s. custom tools that gets executed on server? IIUC, "memory" tools are custom tools that gets executed on server?

Reply:
good catch. changed to UserDefinedToolDef